drivers/interrupt_controller/intc_gicv3.c

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright 2020 Broadcom
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <assert.h>
+#include <sw_isr_table.h>
+#include <dt-bindings/interrupt-controller/arm-gic.h>
+#include <drivers/interrupt_controller/gic.h>
+
+/* TODO: probe for each implemented rdist base */
+#define GIC_RDIST_BASE  DT_REG_ADDR_BY_IDX(DT_INST(0, arm_gic), 1)
+
+#define GICR_SGIBASE_OFFSET	0x10000
+
+/* Offsets from GICD base or GICR sgi_base */
+#define	GIC_CTRL		0x0
+#define	GIC_TYPER		0x4
+#define	GIC_IIDR		0x8
+#define	GIC_IGROUPR		0x80
+#define	GIC_ISENABLER		0x100
+#define	GIC_ICENABLER		0x180
+#define	GIC_ISPENDR		0x200
+#define	GIC_ICPENDR		0x280
+#define	GIC_ISACTIVER		0x300
+#define	GIC_ICACTIVER		0x380
+#define	GIC_IPRIORITYR		0x400
+#define	GIC_ITARGETSR		0x800
+#define	GIC_ICFGR		0xc00
+#define GIC_IGROUPMODR		0xd00
+#define	GIC_SGIR		0xf00
```

Comment:
These offsets are same for GICD and GICR  and hence I am trying to use common logic to get at index register for
corresponding apis eg:  `arm_gic_irq_enable(irq)`   uses ` ISENABLER(base, n)`  to choose right register.  Only base will
very for `SPI/PPI/SGI`.